docs/machine-learning/tutorials/taxi-fare.md

```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
-```
+When the model is trained and evaluated, by default, the values in the **Label** column are considered as correct values to be predicted. As we want to predict the taxi trip fare, copy the `FareAmount` column into the **Label** column. To do that, use the `CopyColumnsEstimator` transformation class, and add the following code:
 
-The last step in data preparation combines all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Legacy.Transforms.ColumnConcatenator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
+[!code-csharp[CopyColumnsEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#7 "Use the CopyColumnsEstimator")]
 
-```csharp
-pipeline.Add(new ColumnConcatenator("Features",
-                                    "VendorId",
-                                    "RateCode",
-                                    "PassengerCount",
-                                    "TripDistance",
-                                    "PaymentType"));
-```
+The algorithm that trains the model requires **numeric** features, so you have to transform the categorical data (`VendorId`, `RateCode`, and `PaymentType`) values into numbers. To do that, use the `OneHotEncodingEstimator` transformation class, which assigns different numeric key values to the different values in each of the columns, and add the following code:
 
-Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. You already determined that it isn't a useful prediction feature.
+[!code-csharp[OneHotEncodingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#8 "Use the OneHotEncodingEstimator")]
 
-> [!NOTE]
-> These steps must be added to the pipeline in the order specified above for successful execution.
+The last step in data preparation combines all of the feature columns into the **Features** column using the `ColumnConcatenatingEstimator` transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
 
-## Choose a learning algorithm
+[!code-csharp[ColumnConcatenatingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#9 "Use the ColumnConcatenatingEstimator")]
 
-After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. You chose a **regression** task for this problem, so you use a <xref:Microsoft.ML.Legacy.Trainers.FastTreeRegressor> learner, which is one of the regression learners provided by ML.NET.
+Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. We already determined that it isn't a useful prediction feature.
```

Comment:
Updated snippets include the "TripTime" column, so this paragraph doesn't match the code:
https://github.com/dotnet/samples/pull/441/files#diff-533141b9bd48d8d1591a35cdef83ab0fR80

Reply:
@pkulikov - Good catch. Thanks!

Reply:
@JRAlexander as this is already merged, another PR should be made to update the line 69. Now, it is:
>**trip_time_in_secs:** The amount of time the trip took. You want to predict the fare of the trip before the trip is completed. At that moment you don't know how long the trip would take. Thus, the trip time is not a feature and you'll exclude this column from the model.